ui/hud/hud.gd

```diff
@@ -63,6 +64,27 @@ func update_keys():
 		$keys.show()
 		$keys/keys.text = str(network.current_map.get_node("dungeon_handler").keys).pad_zeros(1)
 
+func update_buttons():
+	#Checks if there are any controllers connected
+	if Input.get_connected_joypads().size() > 0:
+		# Show controller buttons
+		name = Input.get_joy_name(0)
+		# If controller is XInput, show Xbox buttons
+		if "XInput" or "Xbox" in name:
```

Comment:
Instead of a set of if/else statements, what do you think about placing a `match` statement?
https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_basics.html#match

Reply:
The problem is that I check for a substring in the controller name. For Example, if the name contains Xbox or XInput, then its an Xbox controller. I don't check the entire name, there many variations of the same controller. I don't think you can check substrings in a match statement

Reply:
I see... Fair enough.

Maybe those sections can have the repetitive code moved inside a method call instead?

Something like this maybe?
```gdscript
func set_hud_buttons(texture_name):
    for i in ["B", "X", "Y"]:
        get_node("hud2d/buttons/" + i).texture = preload("res://ui/hud/%s" % texture_name)
```